### ISSUE TITLE
Cannot generate a model with a binary attribute

### ISSUE THREAD

**Issue by devinus** [phoenixframework/phoenix#1578]
### Environment

* Elixir version (elixir -v): 1.2.2
* Phoenix version (mix deps): 1.1.4
* NodeJS version (node -v):
* NPM version (npm -v):
* Operating system: OS X 10.11.3

### Expected behavior

I can generate a model with a `binary` field.

### Actual behavior

```
-> % mix phoenix.gen.model RuleSet rule_set ast:binary
** (Mix) Unknown type `binary` given to generator
```

Relevant lines:

https://github.com/phoenixframework/phoenix/blob/02237cf547ad0de0552983586b5d95c9da3f5473/lib/mix/tasks/phoenix.gen.model.ex#L227
https://github.com/elixir-lang/ecto/blob/master/lib/ecto/type.ex#L170
https://github.com/elixir-lang/ecto/blob/master/lib/ecto/type.ex#L106

**Comment by devinus** [comment]
Looks like this was fixed by https://github.com/phoenixframework/phoenix/pull/1512, so potentially a regression?

**Comment by josevalim** [comment]
We have never released after #1512. :)

**Comment by devinus** [comment]
Ah, that would be why.
